ttnn/cpp/ttnn/operations/data_movement/repeat_interleave/repeat_interleave.hpp

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/tensor/types.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/concat/concat_op.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/repeat/repeat_op.hpp"
+#include "ttnn/operations/core/core.hpp"
+#include "ttnn/operations/data_movement/permute/permute.hpp"
+
+#include <ranges>
+
+namespace ttnn {
+namespace operations {
+namespace data_movement {
+
+namespace utils {
+    Tensor repeat_interleave(const Tensor&, uint32_t, int32_t, const std::optional<MemoryConfig>&);
+}
+
+struct ExecuteRepeatInterleave {
+
+    // # This operation does not support the following cases:
+    // #   - Shape([2[32], 2[32]]) -> repeats = 2, dim = 0
+    // #   - Shape([2[32], 2[32]]) -> repeats = Tensor[1,2], dim = 1
+    static ttnn::Tensor operator()(
```

Comment:
whole operator() can be implemented in c++ with utils::repeat_interleave body

Reply:
@ayerofieiev-tt no need for the cpp file then right?. place the entire repeat_interleave fn in operator()

Reply:
Split in .h and .cpp is still a good thing.
Proper split minimizes rebuild time.

Whenever a header changes, every cpp which directly or indirectly includes it has to be rebuilt.
When the implementation is in .cpp, header is changed less often. Things don't have to be rebuild as much.

Reply:
discussed this with Eyon on today's standup, and he suggested to keep implementation in cpp as well. so i've retained that.

Reply:
Definitely agree on retaining impl in .cpp, but lets remove impl from .hpp here